src/test/java/net/kaczmarzyk/e2e/converter/CalendarE2eTest.java

```diff
@@ -0,0 +1,100 @@
+
+import static net.kaczmarzyk.spring.data.jpa.CustomerBuilder.customer;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * @author Hubert Gotfryd (Tratif sp. z o.o.)
+ */
+public class CalendarE2eTest extends E2eTestBase {
+    @Controller
+    public static class CalendarSpecController {
+
+        @Autowired
+        CustomerRepository customerRepo;
+
+        @RequestMapping(value = "/customers", params = "registeredEqualDay")
+        @ResponseBody
+        public Object findCustomersRegisteredDayEqualWithDefaultConfig(
+                @Spec(path="registrationDate", params="registeredEqualDay", spec=EqualDay.class) Specification<Customer> spec) {
+
+            return customerRepo.findAll(spec);
+        }
+
+        @RequestMapping(value = "/customers", params = "registeredEqualDayCustomConfig")
+        @ResponseBody
+        public Object findCustomersRegisteredDayEqualWithCustomConfigContainingTime(
+                @Spec(path="registrationDate", params="registeredEqualDayCustomConfig", config="yyyy-MM-dd\'T\'HH:mm:ss", spec=EqualDay.class) Specification<Customer> spec) {
+
+            return customerRepo.findAll(spec);
+        }
+    }
+
+    @Test
+    public void findsByEqualDayUsingDefaultConfig() throws Exception {
+        customer("Barry", "Benson")
+                .registrationDate(2014, 3, 16)
+                .build(em);
+        customer("Adam", "Flayman")
+                .registrationDate(2014, 3, 14, 23, 59, 59, 999)
+                .build(em);
+
+        mockMvc.perform(get("/customers")
+                        .param("registeredEqualDay", "2014-03-15")
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$").isArray())
+                .andExpect(jsonPath("$[0].firstName").value("Homer"))
+                .andExpect(jsonPath("$[1].firstName").value("Moe"))
+                .andExpect(jsonPath("$[2]").doesNotExist());
+    }
+
+    @Test
+    public void findsByEqualDayUsingCustomConfigIgnoringTime() throws Exception {
+        customer("Barry", "Benson")
+                .registrationDate(2014, 3, 16)
+                .build(em);
+        customer("Adam", "Flayman")
+                .registrationDate(2014, 3, 14, 23, 59, 59, 999)
+                .build(em);
+
+        mockMvc.perform(get("/customers")
+                        .param("registeredEqualDayCustomConfig", "2014-03-15T12:34:19")
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$").isArray())
+                .andExpect(jsonPath("$[0].firstName").value("Homer"))
+                .andExpect(jsonPath("$[1].firstName").value("Moe"))
```

Comment:
sanity check do Homer and Moe have registration date with tiime?

consider saving additional new customer here in test  so that he is included in the result)